Modules/posixmodule.c

```diff
@@ -1652,56 +1589,59 @@ win32_xstat_impl(const wchar_t *path, struct _Py_stat_struct *result,
            If the latter, we can use attributes_from_dir. */
         DWORD lastError = GetLastError();
         if (lastError != ERROR_ACCESS_DENIED &&
-            lastError != ERROR_SHARING_VIOLATION)
+            lastError != ERROR_SHARING_VIOLATION &&
+            lastError != ERROR_INVALID_PARAMETER)
             return -1;
         /* Could not get attributes on open file. Fall back to
            reading the directory. */
         if (!attributes_from_dir(path, &info, &reparse_tag))
             /* Very strange. This should not fail now */
```

Comment:
This comment itself is very strange. Of course it can fail if you don't have access to the directory, e.g. it's in another user's profile and you're not an administrator.